src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.NoValidDataException;
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final User USER_WITHOUT_DATA = null;
+    private static final User USER_WITH_VALID_DATA = new User("Spider", "1234qw", 18);
+    private static final User USER_THAT_ALREADY_EXIST = new User("Spider", "12345q", 20);
+    private static final User SECOND_USER_WITH_VALID_DATA = new User("SuperMario", "1235qw", 50);
+    private static final User THIRD_USER_WITH_VALID_DATA = new User("IronMan", "1236qw", 45);
+    private static final User USER_WITH_INVALID_LOGIN_LENGTH = new User("Mario", "1234qw", 18);
+    private static final User USER_WITHOUT_LOGIN = new User(null, "1234qw", 18);
+    private static final User USER_WITH_INVALID_CHARACTER_IN_LOGIN
+            = new User("Mario34", "1234qw", 18);
+    private static final User USER_WITH_INVALID_PASSWORD_LENGTH = new User("Spider", "1234q", 18);
+    private static final User USER_WITHOUT_PASSWORD = new User("Spider", null, 18);
+    private static final User USER_WITH_INVALID_CHARACTER_IN_PASSWORD
+            = new User("Spider", "1234q7/_", 18);
+    private static final User USER_WITH_INVALID_AGE = new User("Spider", "1234qw", 17);
+    private static final User USER_WITHOUT_AGE = new User("Spider", "1234qw", null);
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void expectingExceptionWhenUserIsNull() {
+        assertThrows(NoValidDataException.class, () -> {
+            registrationService.register(USER_WITHOUT_DATA);
+        });
+    }
+
+    @Test
+    void checkIfUserWasAdded() {
+        StorageDao storageDao = new StorageDaoImpl();
+        registrationService.register(USER_WITH_VALID_DATA);
+        User actual = storageDao.get(USER_WITH_VALID_DATA.getLogin());
+        User expected = USER_WITH_VALID_DATA;
+        assertEquals(expected,actual);
+        storageDao.remote(USER_WITH_VALID_DATA);
```

Comment:
For this, we can use afterEach annotation and clear our list of users instead of removing one by one all time

Reply:
> For this, we can use afterEach annotation and clear our list of users instead of removing one by one all time

I thought about this approach, but in my case, I need to remove added users only after two tests, so I chose this method.